connaisseur/tests/integration/integration-test.sh

```diff
@@ -51,6 +51,51 @@ else
   echo 'Successfully allowed usage of signed image'
 fi
 
+echo 'Testing edge case of tag defined in both targets and release json file...'
+DEPLOYED_SHA=$(kubectl get pod pod -o yaml | yq e '.spec.containers[0].image' - | sed 's/.*sha256://')
+if [[ "${DEPLOYED_SHA}" != 'c5327b291d702719a26c6cf8cc93f72e7902df46547106a9930feda2c002a4a7' ]]; then
+  echo "Connaisseur substituted wrong image: ${DEPLOYED_SHA}"
+  exit 1
+else
+  echo 'Connaisseur substituted correct image'
+fi
+
+echo 'Testing signed image with designated signer...'
+kubectl run pod2 --image=securesystemsengineering/testimage:special_sig >output.log 2>&1 || true
+NUMBER_OF_VALID_DEPLOYMENTS+=1
+
+if [[ "$(cat output.log)" != 'pod/pod2 created' ]]; then
+  echo 'Failed to allow image signed by designated signer. Output:'
+  cat output.log
+  exit 1
+else
+  echo 'Successfully allowed usage of image signed by designated signer'
+fi
+
+echo 'Testing image with missing designated signer...'
+kubectl run pod3 --image=securesystemsengineering/testimage:non_existant >output.log 2>&1 || true
```

Comment:
Instead of testing whether a delegation hasn't signed a tag that doesn't exist in the first place, we should test if that delegation hasn't signed a tag that actually exists, right? so ideally you should add your signature to `:non_existant` so that a signature theoretically exists, but since i'm set up as required delegation, this will fail.

Reply:
Added it as `:wrong_signer` with a sig of mine